src/sslsocket.cpp

```diff
@@ -126,7 +126,16 @@ const char *SSL_error_string(int ssl_error, int orig_ret)
 
 SSL* SSL_new_client()
 {
-    return SSL_new(sip_trp_ssl_ctx_client);
+    SSL* ssl = SSL_new(sip_trp_ssl_ctx_client);
+
+    // Inject the ServerNameIndication (SNI).
+    // It requires to be a hostname. However, SSL takes whatever we set. 
+    // Ref https://datatracker.ietf.org/doc/html/rfc6066#section-3
+    if (strcmp(remote_ip, remote_host) != 0){
```

Comment:
I prefer `!strcmp(...)`, and please add a space before the brace.

Reply:
on it

Reply:
in my case, `!strcmp` would not be right. I need to set the hostname if it differs from the remote_ip

Reply:
Right, my mistake.